strongbox-web-core/src/main/java/org/carlspring/strongbox/app/boot/BootProgressServlet.java

```diff
@@ -0,0 +1,126 @@
+            { "org.carlspring.strongbox.config.hazelcast.HazelcastConfiguration", "Waiting for cache..." },
+            { "org.springframework.boot.actuate.autoconfigure", "Waiting for actuators..." },
+            // it might look like we need to use "storageBooter" here, but that actually depends on tempDirBooter which does most of the heavy lifting.
+            { "tempDirBooter", "Waiting for storage booter" },
+            { "storageProviderRegistry", "Loading storage providers..." },
+            { "maven2LayoutProvider", "Loading Maven layout provider.." },
+            { "nugetLayoutProvider", "Loading Nuget layout provider.." },
+            { "npmLayoutProvider", "Loading NPM layout provider.." },
+            { "rawLayoutProvider", "Loading Raw layout provider.." },
+            { "transactionManager", "Loading transaction manager..." },
+            { "org.carlspring.strongbox.config.SecurityConfig", "Loading security configuration..." },
+            { "org.carlspring.strongbox.config.SwaggerConfig", "Loading documentation..." },
+            { "fallback", "Waiting for services to go live..." }
+    }).collect(Collectors.toMap(p -> p[0], p -> p[1]));
+    //@formatter:on
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException
+    {
+
+        if (!MediaType.TEXT_EVENT_STREAM_VALUE.equalsIgnoreCase(request.getHeader("Accept")))
+        {
+            response.sendError(HttpStatus.NOT_ACCEPTABLE.value());
+        }
+        else
+        {
+            response.setStatus(HttpStatus.OK.value());
+            response.setContentType(MediaType.TEXT_EVENT_STREAM_VALUE);
+            response.setCharacterEncoding("UTF-8");
+
+            response.setHeader("Cache-control", "no-cache, no-store");
+            response.setHeader("Pragma", "no-cache");
+            response.setHeader("Expires", "-1");
+            // -- CORS - allow everybody while booting, but set max-age force browsers re-check after boot.
+            response.setHeader("Allow", "GET, OPTIONS");
+            response.setHeader("Access-Control-Allow-Origin", "*");
+            response.setHeader("Access-Control-Allow-Headers", "*");
+            response.setHeader("Access-Control-Max-Age", "600");
+
+            response.flushBuffer();
+
+            final AsyncContext asyncContext = request.startAsync();
+
+            final Observable<String> observable = BootProgressBeanPostProcessor.getObservable()
+                                                                               .observeOn(Schedulers.single())
+                                                                               .subscribeOn(Schedulers.io())
+                                                                               .filter(this::filterBeansByName)
+                                                                               .map(this::getBeanDisplayName);
```

Comment:
what about this comment ?

Reply:
can't `Observable<String>` be initialized (`BootProgressBeanPostProcessor#getProgressObservable`) once in `BootProgressBeanPostProcessor#startup` ?

Reply:
I don't think that is possible in this case. You need to subscribe to the observable in order to stream the emitted data to all incoming requests. In order to do that, you need to use the `asyncContext` which is not available in `BootProgressBeanPostProcessor`. 

Ping me in the chat if you think otherwise. :)